forge-game/src/main/java/forge/game/zone/Zone.java

```diff
@@ -78,6 +79,46 @@ public final void reorder(final Card c, final int index) {
         cardList.add(index, c);
     }
 
+    public final void order() {
+        cardList.sort(new Comparator<Card>() {
+            @Override
+            public int compare(final Card firstCard, final Card secondCard) {
+                // if same name then keep together
+                if (firstCard
```

Comment:
use Guava ComparatorChain

```
return ComparisonChain.start()
         .compare(firstCard.getCMC(), secondCard.getCMC())
// add shares color with
         .result();
```

Reply:
you especially need to be careful with such stuff, otherwise you might get like:
`violates its general contract`

when you have 3 cards with same cmc: A, B, C while A shares colors with B, and B with C but A and C doesn't share colors.